### PR TITLE
Euler Traveling Vortex Test

### PR DESCRIPTION
**Pull request by LeilaGhaffari**
This is a WIP PR for the development of the Euler traveling vortex verification test.

### CODE REVIEW

**Comment by jeremylt** [comment]
Note - I recommend rebasing this branch off of main, as the CI infrastructure has been significantly overhauled.

**Comment by LeilaGhaffari** [comment]
> Note - I recommend rebasing this branch off of main, as the CI infrastructure has been significantly overhauled.

Done. Thanks.

File: examples/fluids/navierstokes.c [hunk 1]
```diff
@@ -1831,6 +1833,15 @@ int main(int argc, char **argv) {
...
     ierr = VecAXPY(Q, -1.0, Qexact);  CHKERRQ(ierr);
+    // Visualize the absolute error
+    ierr = PetscSNPrintf(filepath, sizeof filepath, "%s/ns-err-%03D.vtu",
+                         user->outputfolder, user->contsteps);
+    CHKERRQ(ierr);
+    ierr = PetscViewerVTKOpen(PetscObjectComm((PetscObject)Q), filepath,
+                            FILE_MODE_WRITE, &viewer); CHKERRQ(ierr);
+    ierr = VecView(Q, viewer); CHKERRQ(ierr);
+    ierr = PetscViewerDestroy(&viewer); CHKERRQ(ierr);
+    // Compute the relative error
```
Comment (LeilaGhaffari) [inline examples/fluids/navierstokes.c]: I am not sure whether we should keep this or not. @jedbrown , let me know if you think I should drop this commit or make any changes (like visualizing the relative error instead).

Comment (jedbrown) [reply examples/fluids/navierstokes.c]: I don't think it's necessary to keep it, but if you do, please disable it by default (adding an option to activate it).

Comment (LeilaGhaffari) [reply examples/fluids/navierstokes.c]: Thanks! I dropped it.

**Comment by LeilaGhaffari** [comment]
Some sample runs:

**Traveling Vortex**

Implicit scheme:
```
./navierstokes -problem euler_vortex -vortex_strength 5 -ts_adapt_monitor -lx 1000 -ly 1000 -lz 1 -dm_plex_box_faces 20,20,1 -ts_adapt_safety 0.8 -implicit -snes_atol 1e-3 -ksp_atol 1e-4 -ksp_rtol 1e-3 -ts_type alpha -ts_dt 2e-3 -ksp_type bcgs -snes_lag_jacobian 100 -snes_lag_jacobian_persists -snes_mf_operator -viz_refine 2 -problem_euler_mean_velocity 1,1,0 -ts_max_time 4
```

Explicit scheme:
```
./navierstokes -problem euler_vortex -ts_adapt_dt_max .01 -dm_plex_box_faces 40,40,1 -degree 2 -lx 1000 -ly 1000 -lz 1 -problem_euler_mean_velocity 1.,1.,0. -ts_adapt_monitor -ts_max_time 4
```

euler_vortex

**Traveling bubble**
```
./navierstokes -problem euler_vortex  -ts_adapt_dt_max .01 -dm_plex_box_faces 40,40,1 -degree 1 -lx 600 -ly 600 -lz 1 -problem_euler_mean_velocity 1.,-1.,0. -ts_adapt_monitor -ts_max_time 2 -euler_test t4
```
case4

**Comment by LeilaGhaffari** [comment]
This is ready for review.
I can add some regression tests for this problem after [#687](https://github.com/CEED/libCEED/pull/687) is merged (or maybe merging [#687](https://github.com/CEED/libCEED/pull/687) first?!).

**Pull request merged**